verifiable-db/src/cells_tree/api.rs

```diff
@@ -37,27 +37,79 @@ pub enum CircuitInput {
 
 impl CircuitInput {
     /// Create a circuit input for proving a leaf node.
+    /// It is not considered a multiplier column. Please use `leaf_multiplier` for registering a
+    /// multiplier column.
     pub fn leaf(identifier: u64, value: U256) -> Self {
-        CircuitInput::Leaf(LeafCircuit {
-            identifier: F::from_canonical_u64(identifier),
-            value,
-        })
+        CircuitInput::Leaf(
+            Cell {
+                identifier: F::from_canonical_u64(identifier),
+                value,
+                is_multiplier: false,
+            }
+            .into(),
+        )
+    }
+    /// Create a circuit input for proving a leaf node whose value is considered as a multiplier
+    /// depending on the boolean value.
+    /// i.e. it means it's one of the repeated value amongst all the rows
+    pub fn leaf_multiplier(identifier: u64, value: U256, is_multiplier: bool) -> Self {
```

Comment:
my initial thought was more on keeping the previous API the same and it ended up as you said 
>  the API in this way because you want to always use leaf for tables not mixing multiple variable types and leaf_multiplier for mixed tables

If if i set `leaf` to set `multiplier = false` directly then we would not use the `individual` keyword that i've been trying to instill in the rest of the codebase.
So maybe `leaf_merge` is better name ?

Reply:
The renaming would make sense if we want to keep the usage of `leaf_merge` only for merged tables. But for instance in the integration test it seems to me you are using this method for all tests cases, isn't it? So I am wondering if might be always convenient to just have a single method with the multiplier flag rather than having 2 distinct methods

Reply:
@hackaugusto what do you prefer ? :D Any opinion ? I was more on the side of letting things backward compatible and adding new endpoint, it doesn't "surprise" me more than that but dont really have an strong opinion.